src/mon/NVMeofGwMap.cc

```diff
@@ -0,0 +1,585 @@
+        cancel_timer(gw_id, group_key, grpid);
+        for (auto& gw_state: Created_gws[group_key]) {
+            auto& st = gw_state.second;
+            if (st.sm_state[grpid] == GW_STATES_PER_AGROUP_E::GW_OWNER_WAIT_FAILBACK_PREPARED && st.availability == GW_AVAILABILITY_E::GW_AVAILABLE )
+            {
+                fbp_gw_state.standby_state(grpid);// Previous failover GW  set to standby
+                st.active_state(grpid);
+                dout(10)  << "Expired Failback-preparation timer from GW " << gw_id << " ANA groupId "<< grpid << dendl;
+                map_modified = true;
+                break;
+            }
+            else if (st.ana_grp_id == grpid){
+                if(st.sm_state[grpid] == GW_STATES_PER_AGROUP_E::GW_STANDBY_STATE  &&  st.availability == GW_AVAILABILITY_E::GW_AVAILABLE) {
+                    st.active_state(grpid);// GW failed and started during the persistency interval
+                    dout(4)  << "Failback unsuccessfull. GW: " << gw_state.first << "becomes Active for the ana group " << grpid  << dendl;
+                }
+                fbp_gw_state.standby_state(grpid);
+                dout(4) << "Failback unsuccessfull GW: " << gw_id << "becomes standby for the ana group " << grpid  << dendl;
+                map_modified = true;
+                break;
+            }
+        }
+    }
+    else if(fbp_gw_state.sm_state[grpid] == GW_STATES_PER_AGROUP_E::GW_WAIT_BLOCKLIST_CMPL){
+        dout(1) << " Expired GW_WAIT_FAILOVER_PREPARED timer from GW " << gw_id << " ANA groupId: "<< grpid << dendl;
+        ceph_assert(false);
+    }
+}
+
+NvmeGwCreated& NVMeofGwMap::find_already_created_gw(const NvmeGwId &gw_id, const NvmeGroupKey& group_key)
+{
+    auto& group_gws = Created_gws[group_key];
+    auto  gw_it = group_gws.find(gw_id);
+    ceph_assert(gw_it != group_gws.end());//should not happen
+    return gw_it->second;
+}
+
+struct CMonRequestProposal : public Context {
+  NVMeofGwMap *m;
+  CMonRequestProposal(NVMeofGwMap *mon) : m(mon) {}
+  void finish(int r) {
+      dout(4) << "osdmon is  writable? " << m->mon->osdmon()->is_writeable() << dendl;
+      if(m->mon->osdmon()->is_writeable()){
+        m->mon->nvmegwmon()->request_proposal(m->mon->osdmon());
+      }
+      else {
+          m->mon->osdmon()->wait_for_writeable_ctx( new CMonRequestProposal(m));
+      }
+  }
+};
+
+int NVMeofGwMap::blocklist_gw(const NvmeGwId &gw_id, const NvmeGroupKey& group_key, NvmeAnaGrpId grpid, epoch_t &epoch, bool failover)
+{
+    NvmeGwCreated& gw_map =  Created_gws[group_key][gw_id];  //find_already_created_gw(gw_id, group_key);
+
+     if (gw_map.nonce_map[grpid].size() > 0){
+        NvmeNonceVector &nonce_vector = gw_map.nonce_map[grpid];;
+        std::string str = "[";
+        entity_addrvec_t addr_vect;
+        //auto until = ceph_clock_now();  // until += g_conf().get_val<double>("mon_mgr_blocklist_interval"); // 1day - TODO
+        utime_t until(600, 0);// 1hour delay
```

Comment:
And `600, 0` is 10 minutes, not 1 hour as the comment says...

The comment is probably from my previous suggestion to make it 1 hour based on `mon_osd_blocklist_default_expire` default: https://github.com/ceph/ceph/pull/54671#discussion_r1522113475.  Instead of hard coding and getting yourself exposed to typos like this, I'd recommend querying `mon_osd_blocklist_default_expire` value like it's done in `OSDMonitor` for `ceph osd blocklist add` command.

Reply:
The `... "mon_mgr_blocklist_interval"); // 1day - TODO` comment above should go away too.